flytepropeller/pkg/apis/flyteworkflow/v1alpha1/node_status.go

```diff
@@ -316,6 +316,27 @@ func (in *ArrayNodeStatus) SetTaskPhaseVersion(taskPhaseVersion uint32) {
 	}
 }
 
+func (in *ArrayNodeStatus) DeepCopyInto(out *ArrayNodeStatus) {
+	*out = *in
+	out.MutableStruct = in.MutableStruct
+
+	if in.ExecutionError != nil {
+		in, out := &in.ExecutionError, &out.ExecutionError
+		*out = new(core.ExecutionError)
+		*out = *in
+	}
+}
+
+func (in *ArrayNodeStatus) DeepCopy() *ArrayNodeStatus {
+	if in == nil {
+		return nil
+	}
+
+	out := &ArrayNodeStatus{}
```

Comment:
Why are we copying into an empty `ArrayNodeStatus`?

Reply:
This method returns a deep copy of the ArrayNodeStatus.

This is similar to [func (in *TaskNodeStatus) DeepCopy() *TaskNodeStatus](https://github.com/flyteorg/flyte/blob/38cc76dd78e7f46283f160189cb6b38cf8fb1c8d/flytepropeller/pkg/apis/flyteworkflow/v1alpha1/node_status.go#L999)

Reply:
sorry, got the direction wrong. Thank for clarifying!